overlord/hookstate/hookmgr.go

```diff
@@ -216,8 +218,13 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 
 	if hookExists {
 		output, err := runHook(context, tomb)
-		if err != nil {
+		if hooksup.IgnoreFail && err != nil {
+			task.State().Lock()
+			task.Logf("ignoring failure in hook %q: %v", hooksup.Hook, err)
+			task.State().Unlock()
+		} else if err != nil {
 			err = osutil.OutputErr(output, err)
```

Comment:
You mean in this error path here? https://github.com/snapcore/snapd/pull/3071/files/a27f715bda5bc5cae149282695a92a6c126f0848#diff-87319759a99457f870b9a8a989aa7867L229 ? I can add it there, yes, but for our case it looks like the handlers for Done() set in configstate is always returning nil. I could rename "ignoreFail" to "ignoreHookFail" maybe? Or maybe I am missing a bit of the bigger picture here :)